### ISSUE TITLE
TimeDistributed does not propagate mask and prevents inner Sequential from propagating mask

### ISSUE THREAD

**Issue by mpdn** [tensorflow/tensorflow#31684]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 1.14.0, 2.0.0b
- Python version: 3.6.8

**Code to reproduce the issue**
This fails due to failed assertion:

```
import tensorflow as tf

def check_mask(inputs, mask=None):
    assert mask is not None
    return inputs

model = tf.keras.layers.TimeDistributed(tf.keras.models.Sequential([
    tf.keras.layers.Masking(0),
    tf.keras.layers.Lambda(check_mask)
]))

model(tf.convert_to_tensor([[[0]]]))
```

I would expect the mask to be propagated, as is normal in `Sequential`. The `TimeDistributed` seems to somehow prevent the inner `Sequential` from doing so.

This also fails:
```
import tensorflow as tf

def check_mask(inputs, mask=None):
    assert mask is not None
    return inputs

model = tf.keras.models.Sequential([
    tf.keras.layers.TimeDistributed(tf.keras.layers.Masking(0)),
    tf.keras.layers.TimeDistributed(tf.keras.layers.Lambda(check_mask)),
])

model(tf.convert_to_tensor([[[0]]]))
```

**Other info / logs**

This might be related to another issue I also submitted: https://github.com/tensorflow/tensorflow/issues/31638

**Comment by gadagashwini** [comment]
I am able to replicate the issue with Tensorflow 2.0.0.beta1 on colab. Please take a look at gist [here](https://colab.research.google.com/drive/1z8DpMed2ft6Wxdq-79ldjo0ruZtAxia4). Thanks!

**Comment by yhliang2018** [comment]
In sequential and functional models, mask is propagated by the `_keras_mask` property (check [here](https://github.com/tensorflow/tensorflow/blob/1933f1b166cc141d9f85de1ff3772f96c2c3760c/tensorflow/python/keras/engine/sequential.py#L297) for the example in sequential model). However, the explicitly specified arg `mask` has higher priority than the private attribute `_keras_mask`. So you will always get `None` in the `check_mask` method as it's specified as `None` explicitly. 

You may check [this example](url) to see how `https://github.com/tensorflow/tensorflow/blob/0659606b82c81516d52c2f449b7929b5d5e76c79/tensorflow/python/keras/layers/wrappers_test.py#L257` layer works with mask.
